fineract-provider/src/main/java/org/apache/fineract/portfolio/creditscorecard/domain/CreditScorecardFeature.java

```diff
@@ -0,0 +1,166 @@
+            org.apache.fineract.portfolio.creditscorecard.domain.FeatureDataType dataType,
+            org.apache.fineract.portfolio.creditscorecard.domain.FeatureCategory category, Boolean active) {
+        this.name = name;
+        this.valueType = valueType.getValue();
+        this.dataType = dataType.getValue();
+        this.category = category.getValue();
+        this.active = active;
+    }
+
+    public static CreditScorecardFeature fromJson(JsonCommand command) {
+        final String name = command.stringValueOfParameterNamed("name");
+        final org.apache.fineract.portfolio.creditscorecard.domain.FeatureValueType valueType = org.apache.fineract.portfolio.creditscorecard.domain.FeatureValueType
+                .fromInt(command.integerValueOfParameterNamed("valueType"));
+        final org.apache.fineract.portfolio.creditscorecard.domain.FeatureDataType dataType = org.apache.fineract.portfolio.creditscorecard.domain.FeatureDataType
+                .fromInt(command.integerValueOfParameterNamed("dataType"));
+        final org.apache.fineract.portfolio.creditscorecard.domain.FeatureCategory category = org.apache.fineract.portfolio.creditscorecard.domain.FeatureCategory
+                .fromInt(command.integerValueOfParameterNamed("category"));
+        final boolean active = command.booleanPrimitiveValueOfParameterNamed("active");
+
+        return new CreditScorecardFeature(name, valueType, dataType, category, active);
+    }
+
+    public String getName() {
+        return this.name;
+    }
+
+    public EnumOptionData getValueType() {
+        return CreditScorecardEnumerations.featureValueType(valueType);
+    }
+
+    public EnumOptionData getDataType() {
+        return CreditScorecardEnumerations.featureDataType(dataType);
+    }
+
+    public EnumOptionData getCategory() {
+        return CreditScorecardEnumerations.featureCategory(category);
+    }
+
+    public boolean isActive() {
+        return this.active;
+    }
+
+    public boolean isDeleted() {
+        return this.deleted;
+    }
+
+    /**
+     * Delete is a <i>soft delete</i>. Updates flag on feature so it wont appear in query/report results.
+     *
+     * Any fields with unique constraints and prepended with id of record.
+     */
+    public void delete() {
+        this.deleted = true;
+        this.name = getId() + "_" + this.name;
+    }
+
+    public CreditScorecardFeatureData toData() {
+        final EnumOptionData valueType = CreditScorecardEnumerations.featureValueType(this.valueType);
+        final EnumOptionData dataType = CreditScorecardEnumerations.featureDataType(this.dataType);
+        final EnumOptionData category = CreditScorecardEnumerations.featureCategory(this.category);
+
+        return CreditScorecardFeatureData.instance(null, getId(), this.name, valueType, dataType, category, this.active, null, null, null,
```

Comment:
Can't we write a overloaded method for `CreditScorecardFeatureData.instance` instead of passing many `null`s? Just an idea :)